File: internal/indexer/testdata/rust/views/critical_user_model_view.rs
use crate::models::critical_user_model::CriticalUserModel;

/// CriticalUserModelRenderer renders a CriticalUserModel for display.
pub struct CriticalUserModelRenderer {
    pub model: CriticalUserModel,
}

/// render_critical_user_model formats a CriticalUserModel as a display string.
pub fn render_critical_user_model(m: &CriticalUserModel) -> String {
    format!("{} <{}>", m.name, m.email)
}

/// create_critical_user_model creates a CriticalUserModel with the given name and email.
pub fn create_critical_user_model(name: &str, email: &str) -> CriticalUserModel {
    CriticalUserModel::new(name, email)
}

File: internal/indexer/testdata/go/views/critical_user_model_view.go
package views

import "models"

// CriticalUserModelRenderer renders a CriticalUserModel for display.
type CriticalUserModelRenderer struct {
	Model models.CriticalUserModel
}

// RenderCriticalUserModel formats a CriticalUserModel as a display string.
func RenderCriticalUserModel(m models.CriticalUserModel) string {
	return m.Name + " <" + m.Email + ">"
}

// CreateCriticalUserModelView creates a renderer for a CriticalUserModel.
func CreateCriticalUserModelView(name, email string) models.CriticalUserModel {
	return models.CriticalUserModel{Name: name, Email: email}
}
